### PR TITLE
Generate translations locally for e2e tests

### LINKED ISSUE #50910: Explore building our own translation po/mo/json instead of off the shelf translation in E2E
This is a follow up from https://github.com/woocommerce/woocommerce/pull/50892. In our E2E tests, we load up NL_nl translations to test against, this has some drawbacks:

- We're always testing against the previous release translation, for code currently in trunk, this means we run into a lot of false negatives (which causes us to disable E2E tests until further release). https://github.com/woocommerce/woocommerce/pull/45371
- We avoid false positives that caused the bug linked above (#50892), because migration might still work in a current release but stop working in a future release (false positives).

@dinhtungdu did explore this once and I suggest we go this route again.

### PR DESCRIPTION
**Pull request by alexflorisca**
### Changes proposed in this Pull Request:

This PR is a revival of https://github.com/woocommerce/woocommerce-blocks/pull/6424 to get translation tests working without having to download any languages. Big thanks to @dinhtungdu who did most of the work here and helped me get this over the line. We have something that works, at least for javascript files.

We're generating the `.json` translation files locally and pointing WP to use these. I've had to remove one string `Your cart is currently empty!` from the check as this is generate php side. However inspecting the environment on trunk, it's not translated either so I'm not quite sure how these tests are passing at the moment. I think you'd have to re-add the cart block once you change language, as the title is stored as an attribute in the db.

The caveat to this approach is that it doesn't work for php, only js generated translations. I think this is ok as a first pass, and if and when we need to translate php strings, we can build on top of it.

Closes #50910.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

Tests passing on this branch should be sufficient, but if you want to check locally:

1. Checkout this branch and run `npm run wp-env destroy && npm run env:start` from `plugins/woocommerce-blocks`. Check that there area bunch of `.json` translation files generated in `plugins/woocommerce/i18n/languages`. 
2. Run the following command to run all translation tests  `npm run test:e2e -- mini-cart-block.shopper.block_theme.spec.ts cart-checkout-block-translations.shopper.block_theme.spec.ts --ui`

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [x] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

This PR updates some tests, no changelog required.

</details>

### CODE REVIEW

File: plugins/woocommerce-blocks/tests/e2e/tests/cart/cart-checkout-block-translations.shopper.block_theme.spec.ts
Comment (alexflorisca) [inline plugins/woocommerce-blocks/tests/e2e/tests/cart/cart-checkout-block-translations.shopper.block_theme.spec.ts:70]: Duplicate of the above

File: plugins/woocommerce-blocks/tests/e2e/tests/cart/cart-checkout-block-translations.shopper.block_theme.spec.ts
Comment (alexflorisca) [inline plugins/woocommerce-blocks/tests/e2e/tests/cart/cart-checkout-block-translations.shopper.block_theme.spec.ts:73]: We don't need to test the translation of the Add to Cart button

File: plugins/woocommerce-blocks/tests/e2e/tests/mini-cart/mini-cart-block.shopper.block_theme.spec.ts
Comment (alexflorisca) [inline plugins/woocommerce-blocks/tests/e2e/tests/mini-cart/mini-cart-block.shopper.block_theme.spec.ts:26]: A random beanie with Logo was being added to the cart sometime before this test so best to empty the cart before each of the tests. This might solve some flakiness, who knows 🤷

**Review by wavvves — COMMENTED** [review]
I think the changes affected expected values on tests, multiple CI jobs are failing due to wrong counts 🤔

**Comment by alexflorisca** [comment]
Pushed a fix that should solve the test failures and tidy up some of the scripts. Will take a look tomorrow if still failing, otherwise should be good to review 🙏

File: plugins/woocommerce-blocks/bin/wp-env-config.sh
Comment (alexflorisca) [inline plugins/woocommerce-blocks/bin/wp-env-config.sh:1]: The `wp-env-config.sh` and `wp-env-pre-config.sh` were left over config files from before we migrated tests to playwright. Now the test config lives in `test-env-setup` and related directories.

File: plugins/woocommerce-blocks/tests/e2e/bin/scripts/parallel/languages.sh
Comment (alexflorisca) [inline plugins/woocommerce-blocks/tests/e2e/bin/scripts/parallel/languages.sh:1]: We generate languages outside of wp-env, so this isn't needed any more

File: plugins/woocommerce-blocks/tests/e2e/test-data/data/data.ts
Comment (alexflorisca) [inline plugins/woocommerce-blocks/tests/e2e/test-data/data/data.ts:1]: This files needs to be a script rather than a module so that it can be run from the command line with node.

File: plugins/woocommerce/package.json
Comment (alexflorisca) [inline plugins/woocommerce/package.json]: Left these in, because they're generated for every PR now so I think they should be on trunk

File: plugins/woocommerce-blocks/tests/e2e/bin/scripts/parallel/languages.sh
Comment (wavvves) [reply plugins/woocommerce-blocks/tests/e2e/bin/scripts/parallel/languages.sh:1]: Maybe we're missing something, there's[ CI jobs](https://github.com/woocommerce/woocommerce/actions/runs/11721011507/job/33615015804?pr=52188) complaining about this language not being installed 🤔

File: plugins/woocommerce-blocks/tests/e2e/bin/scripts/parallel/languages.sh
Comment (wavvves) [inline plugins/woocommerce-blocks/tests/e2e/bin/scripts/parallel/languages.sh:4]: Maybe we're missing something, there's[ CI jobs](https://github.com/woocommerce/woocommerce/actions/runs/11721011507/job/33615015804?pr=52188) complaining about this language not being installed 🤔

**Review by wavvves — COMMENTED** [review]
Hey @alexflorisca there seems to be some issues around nl_NL not being installed

File: plugins/woocommerce-blocks/tests/e2e/bin/scripts/parallel/languages.sh
Comment (alexflorisca) [reply plugins/woocommerce-blocks/tests/e2e/bin/scripts/parallel/languages.sh:1]: Yep my bad, the core language is still needed, the plugin ones isn't. RE-added languages.sh 👍

File: plugins/woocommerce-blocks/bin/wp-env-config.sh
Comment (wavvves) [inline plugins/woocommerce-blocks/bin/wp-env-config.sh:34]: Following the testing steps it complained about missing script `wp-env:config` after generating translations:
```bash
Success: Updated 'woocommerce_coming_soon' option.
✔ Ran `wp option update woocommerce_coming_soon no` in 'tests-cli'. (in 1s 146ms)
Generating test translations
npm ERR! Missing script: "wp-env:config"
```

I see this was removed, and thought the testing steps could be outdated, but when I saw this file contents I was left wondering if this part was removed by accident:

```shell
if [ $? -eq 0 ]
then
	EXIT_CODE=0
else
  wp user create customer customer@woocommercecoree2etestsuite.com --user_pass=password --role=customer
  EXIT_CODE=$?
fi

## set permalinks for easier wp-json
wp rewrite structure '/%postname%/' --hard
wp core version --extra
wp plugin list
wp theme activate storefront
wp wc customer update 1 --user=1 --billing='{"first_name":"John","last_name":"Doe","company":"Automattic","country":"US","address_1":"addr 1","address_2":"addr 2","city":"San Francisco","state":"CA","postcode":"94107","phone":"123456789"}' --shipping='{"first_name":"John","last_name":"Doe","company":"Automattic","country":"US","address_1":"addr 1","address_2":"addr 2","city":"San Francisco","state":"CA","postcode":"94107","phone":"123456789"}'
```

Can you confirm this wasn't a mistake?
Also is there any chance that removing `wp-env:config` might break any other automations we have besides the CI jobs on this PR? 🤔

**Review by wavvves — COMMENTED** [review]
Hey Alex, everything checks out, but I left a question before I can approve it sorry 🙏

File: plugins/woocommerce-blocks/bin/wp-env-config.sh
Comment (alexflorisca) [reply plugins/woocommerce-blocks/bin/wp-env-config.sh:34]: This wasn't a mistake, the testing instructions were outdated, I've updated them. But this did point to some redundant scripts that were still relying on `wp-env:config`, so I've removed those in https://github.com/woocommerce/woocommerce/pull/52188/commits/6e0c7aee2f0af647654badb98676998435e97a61

Comment (wavvves) [reply plugins/woocommerce-blocks/bin/wp-env-config.sh:34]: I'm not referring to `wp-env:config` calls within our codebase, but I'll guess we'll find out on the betas and RC if any process we have eyes on breaks. But third-party agencies might have this in their pipelines, and although it is their responsibility to adjust, we probably should keep the command, just changing it into echoing a deprecated notice. And also keep the `wp-env-config.sh` doing the deprecated echo there in case someone is calling it directly.
This way we ensure we warn before breaking.

Also I'm not sure if I expressed myself correctly, to be clear, is this part redundant and safe to remove?

```bash
## set permalinks for easier wp-json
wp rewrite structure '/%postname%/' --hard
wp core version --extra
wp plugin list
wp theme activate storefront
wp wc customer update 1 --user=1 --billing='{"first_name":"John","last_name":"Doe","company":"Automattic","country":"US","address_1":"addr 1","address_2":"addr 2","city":"San Francisco","state":"CA","postcode":"94107","phone":"123456789"}' --shipping='{"first_name":"John","last_name":"Doe","company":"Automattic","country":"US","address_1":"addr 1","address_2":"addr 2","city":"San Francisco","state":"CA","postcode":"94107","phone":"123456789"}'
```

Comment (alexflorisca) [reply plugins/woocommerce-blocks/bin/wp-env-config.sh:34]: I get what you're saying but both `wp-env:config` and the scripts I removed are not used anymore. They relate to old e2e tests before we migrated to playwright. So it doesn't do anything. I also don't see why any agencies would be running wc tests in the first place, they'll likely be running their own tests on their plugins/themes. I think it's safe to delete, but maybe let's get a second opinion here from @woocommerce/rubik?

> Also I'm not sure if I expressed myself correctly, to be clear, is this part redundant and safe to remove?

Yes, this setup now happens in `tests/e2e/bin/scripts`.

File: plugins/woocommerce-blocks/package.json
Comment (wavvves) [inline plugins/woocommerce-blocks/package.json:102]:
```suggestion
		"wp-env:config": "echo \"⚠️  WARNING: 'wp-env:config' is deprecated and will be removed in the future.\""
```

**Review by wavvves — APPROVED** [review]
Thanks Alex, approved!

**Pull request merged**
